Amazon.IonDotnet.Tests/Internals/ReaderLocalTableTest.cs

```diff
@@ -48,5 +56,242 @@ void assertTable(ISymbolTable tab, params (string sym, int id)[] syms)
+                    "  imports:" + ION_SYMBOL_TABLE + "," +
+                    "  symbols:[ \"s3\"]" +
+                    "}\n" +
+                LocalSymbolTablePrefix +
+                    "{" +
+                    "  imports:" + ION_SYMBOL_TABLE + "," +
+                    "  symbols:[\"s4\", \"s5\"]" +
+                    "}\n" +
+                LocalSymbolTablePrefix +
+                    "{" +
+                    "  imports:" + ION_SYMBOL_TABLE + "," +
+                    "  symbols:[\"s6\"]" +
+                    "}\n" +
+                    "null";
+
+            ISymbolTable symbolTable = OneValue(text);
+            CheckLocalTable(symbolTable);
+            int systemMaxId = symbolTable.GetSystemTable().MaxId;
+
+            // table contains all symbols and SIDs are in correct order
+            CheckSymbol("s1", systemMaxId + 1, symbolTable);
+            CheckSymbol("s2", systemMaxId + 2, symbolTable);
+            CheckSymbol("s3", systemMaxId + 3, symbolTable);
+            CheckSymbol("s4", systemMaxId + 4, symbolTable);
+            CheckSymbol("s5", systemMaxId + 5, symbolTable);
+            CheckSymbol("s6", systemMaxId + 6, symbolTable);
+
+            CheckUnknownSymbol("unknown", UNKNOWN_SYMBOL_ID, symbolTable);
+            CheckUnknownSymbol(33, symbolTable);
+        }
+
+        [TestMethod]
+        public void TestLocalSymbolTableAppendEmptyList()
+        {
+            string original =
+                LocalSymbolTablePrefix +
+                    "{" +
+                    "  symbols:[ \"s1\"]" +
+                    "}\n";
+
+            string appended = original +
+                LocalSymbolTablePrefix +
+                    "{" +
+                    "  imports:" + ION_SYMBOL_TABLE + "," +
+                    "  symbols:[]" +
+                    "}\n";
+
+            ISymbolTable originalSymbolTable = OneValue(original + "null");
+            ISymbolTable appendedSymbolTable = OneValue(appended + "null") ;
+
+            SymbolToken originalSymbol = originalSymbolTable.Find("s1");
+            SymbolToken appendedSymbol = appendedSymbolTable.Find("s1");
+
+            Assert.AreEqual(originalSymbol.Sid, appendedSymbol.Sid);
+        }
+
+        [TestMethod]
+        public void TestLocalSymbolTableAppendImportBoundary()
+        {
+            string text =
+                LocalSymbolTablePrefix +
+                    "{" +
+                    "  symbols:[ \"s11\"]" +
+                    "}\n" +
+                    "1\n";
+
+            string text2 = 
+                LocalSymbolTablePrefix +
+                    "{" +
+                    "  imports:" + ION_SYMBOL_TABLE + "," +
+                    "  symbols:[ \"s21\"]" +
+                    "}\n" +
+                    "null";
+
+            ISymbolTable original = OneValue(text);
+            ISymbolTable appended = OneValue(text2);
+
+            //TODO this is not implemented, the test could be failing because of this
```

Comment:
Is this comment still relevant?

Reply:
This is the test that is failing and I think it is because it calls Intern but we haven't implemented it.